src/sbvr-api/hooks.ts

```diff
@@ -31,23 +30,27 @@ export interface HookReq {
 	hooks?: InstantiatedHooks;
 	is?: (type: string | string[]) => string | false | null;
 }
-export interface HookArgs {
+export interface HookArgs<Vocab extends string = string> {
 	req: HookReq;
 	request: ODataRequest;
-	api: PinejsClient;
+	api: (typeof api)[Vocab];
 	tx?: Tx | undefined;
 }
 export type HookResponse = PromiseLike<any> | null | void;
 
-export interface Hooks {
-	PREPARSE?: (options: Omit<HookArgs, 'request' | 'api'>) => HookResponse;
-	POSTPARSE?: (options: HookArgs) => HookResponse;
-	PRERUN?: (options: HookArgs & { tx: Tx }) => HookResponse;
+export interface Hooks<Vocab extends string = string> {
+	PREPARSE?: (
+		options: Omit<HookArgs<Vocab>, 'request' | 'api'>,
+	) => HookResponse;
+	POSTPARSE?: (options: HookArgs<Vocab>) => HookResponse;
+	PRERUN?: (options: HookArgs<Vocab> & { tx: Tx }) => HookResponse;
 	/** These are run in reverse translation order from newest to oldest */
-	POSTRUN?: (options: HookArgs & { tx: Tx; result: any }) => HookResponse;
+	POSTRUN?: (
+		options: HookArgs<Vocab> & { tx: Tx; result: any },
```

Comment:
Couldn't result be inferred now? (not necessarily in this PR, just one thing we could keep in mind)

Reply:
Yes, although it's not related to the OData result directly: https://github.com/balena-io/pinejs/pull/765